doc/languages-frameworks/lisp.section.md

```diff
@@ -0,0 +1,225 @@
+# Lisp {#lisp}
+
+## Overview {#lisp-overview}
+
+The Lisp infrastructure works in the following way:
+
+1. Package definitions are imported from Quicklisp (`ql-import.lisp`)
+2. Missing native libraries etc. are fixed in `ql.nix`
+3. Additional packages are defined using `build-asdf-system` in `packages.nix`
+4. CL implementations are enhanced with the `pkgs` and `withPackages` attributes (`makeLisp`)
+5. Packages are overridden individually using `overrideLispAttrs`
+6. Packages are woven into a new scope using `packageOverlays`
+
+## Importing from Quicklisp {#lisp-importing-from-quicklisp}
+
+### The `ql-import` script {#lisp-ql-import}
+
+There is a Common Lisp program in the `import` directory, which downloads the
+`systems.txt` and `releases.txt` files from `quicklisp.org` and creates a Nix
+expression with package definitions in a file called `imported.nix`.
```

Comment:
This trusts Quicklisp dependency data, right?

Reply:
It does trust the QL data - any manual fixes (missing native libraries, extra ASDF systems to compile, patches) can be put in ql.nix or packages.nix.

In practice, most packages imported like that just work, the most common issue is missing native libraries.

Reply:
I think it's the other dimension: the current do-not-trust-QL-upstream-completely system was implemented after some _versions_ of Quicklisp metadata broke _a lot_ of packages.

Reply:
We had a different experience. I haven't seen any breakage in 3 last releases. Though it doesn't mean a future release won't break things. Do you think this is a show-stopper for this patch?

Reply:
Yeah, it's more like on the timescale of a few years than of a few releases that weird versions are sometimes seen (maybe never again in the future if we are lucky…)

If it is a replacement, it needs to at least document that unlike the previous approaches it currently does trust the upstream. I guess if it's needed again, the dependency extraction code could be ported over.